### PR TITLE
accel: add ocp upgrade test

### PR DESCRIPTION
**Pull request by siserafin**
Adding ocp upgrade test case in accel folder

### CODE REVIEW

**Comment by kononovn** [comment]
@siserafin 
accelparams is on ./tests/accel/upgrade level. Please either move it to **./tests/accel/internal/accelparams**  or move it to ./tests/accel/upgrade/internal/tsparams.
You need to think if this parameters are relevant only for upgrade suite or for entire accel level

File: tests/accel/upgrade/internal/url/url.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
+package url
```
Comment (kononovn) [inline tests/accel/upgrade/internal/url/url.go:1]: This package is avaliable in /tests/assisted/internal/url/url.go
Please move it to upper level ./tests/internal/url.go and re-use in your code.

File: tests/accel/upgrade/upgrade.go [hunk 2]
```diff
@@ -0,0 +1,172 @@
+package upgrade
```
Comment (kononovn) [inline tests/accel/upgrade/upgrade.go:1]: Please move it to ./tests/accel/upgrade/tests directory

File: tests/accel/upgrade/workload.go [hunk 3]
```diff
@@ -0,0 +1,215 @@
+package upgrade
```
Comment (kononovn) [inline tests/accel/upgrade/workload.go:1]: Please move it to ./tests/accel/upgrade/tests directory

File: tests/accel/upgrade/upgradeconfig/config.go [hunk 4]
```diff
@@ -0,0 +1,45 @@
+package upgradeconfig
```
Comment (kononovn) [inline tests/accel/upgrade/upgradeconfig/config.go:1]: You need to create ./tests/accel/internal/accelinittools as well as ./tests/accel/internal/accelconfig on level above.

**Review by kononovn — CHANGES_REQUESTED** [review]
Please fix initial comments related to project structure

File: tests/accel/upgrade/internal/url/url.go (hunk 1 above)
Comment (siserafin) [reply tests/accel/upgrade/internal/url/url.go]: Done, thanks

File: tests/accel/upgrade/upgrade.go (hunk 2 above)
Comment (siserafin) [reply tests/accel/upgrade/upgrade.go]: Done, thanks

File: tests/accel/upgrade/upgradeconfig/config.go (hunk 4 above)
Comment (siserafin) [reply tests/accel/upgrade/upgradeconfig/config.go]: Done, thanks

File: tests/accel/internal/accelconfig/config.go [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+	HubKubeConfig        string `envconfig:"ECO_ACCEL_UPGRADE_HUB_KUBECONFIG"`
+	HubClusterName       string `envconfig:"ECO_ACCEL_UPGRADE_HUB_CLUSTER_NAME"`
+	HubAPIClient         *clients.Settings
+}
+
+// NewModulesConfig returns instance of ModulesConfig type.
+func NewModulesConfig() *ModulesConfig {
+	log.Print("Creating new ModulesConfig")
+
+	modulesConfig := new(ModulesConfig)
```
Comment (kononovn) [inline tests/accel/internal/accelconfig/config.go:26]: Please replace by:
```
var modulesConfig ModulesConfig
modulesConfig.GeneralConfig = config.NewConfig()
```

File: tests/accel/internal/accelconfig/config.go [hunk 6]
```diff
@@ -0,0 +1,45 @@
+package upgradeconfig
```
Comment (kononovn) [inline tests/accel/internal/accelconfig/config.go:1]: package name has to be accelconfig

File: tests/accel/internal/accelconfig/config.go [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+	"log"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+
+	"github.com/kelseyhightower/envconfig"
+)
+
+// ModulesConfig contains environment information related to ocp upgrade tests.
+type ModulesConfig struct {
```
Comment (kononovn) [inline tests/accel/internal/accelconfig/config.go:13]: We need to change it to AccelConfig

File: tests/accel/internal/accelconfig/config.go [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+	PullSecret           string `envconfig:"ECO_ACCEL_UPGRADE_PULL_SECRET"`
```
Comment (kononovn) [inline tests/accel/internal/accelconfig/config.go:14]: Are these vars related to UPGRADE test suite only or for all accel area?

File: tests/accel/internal/accelconfig/config.go [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+		glog.V(90).Infof("Creating spoke api client from %s", modulesConfig.HubKubeConfig)
+
+		if modulesConfig.HubAPIClient = clients.New(
+			modulesConfig.HubKubeConfig); modulesConfig.HubAPIClient == nil {
+			glog.V(90).Infof("failed to load provided hub kubeconfig: %v", err)
+		}
+	}
+
+	return modulesConfig
+}
```
Comment (kononovn) [inline tests/accel/internal/accelconfig/config.go:45]: Please review suggested change:


```
package accelconfig

import (
	"log"

	"github.com/golang/glog"
	"github.com/openshift-kni/eco-goinfra/pkg/clients"
	"github.com/openshift-kni/eco-gotests/tests/internal/config"

	"github.com/kelseyhightower/envconfig"
)

// AccelConfig contains environment information related to ocp upgrade tests.
type AccelConfig struct {
	PullSecret           string `envconfig:"ECO_ACCEL_UPGRADE_PULL_SECRET"`
	Registry             string `envconfig:"ECO_ACCEL_UPGRADE_REGISTRY"`
	UpgradeTargetVersion string `envconfig:"ECO_ACCEL_UPGRADE_UPGRADE_TARGET_IMAGE"`
	HubKubeConfig        string `envconfig:"ECO_ACCEL_UPGRADE_HUB_KUBECONFIG"`
	HubClusterName       string `envconfig:"ECO_ACCEL_UPGRADE_HUB_CLUSTER_NAME"`
	HubAPIClient         *clients.Settings
	*config.GeneralConfig
}

// NewAccelConfig returns instance of ModulesConfig type.
func NewAccelConfig() *AccelConfig {
	log.Print("Creating new ModulesConfig")

	var accelConfig *AccelConfig
	accelConfig.GeneralConfig = config.NewConfig()

	err := envconfig.Process("eco_accel_", accelConfig)
	if err != nil {
		log.Printf("failed to instantiate ModulesConfig: %v", err)

		return nil
	}

	if accelConfig.HubKubeConfig != "" {
		glog.V(90).Infof("Creating spoke api client from %s", accelConfig.HubKubeConfig)

		if accelConfig.HubAPIClient = clients.New(
			accelConfig.HubKubeConfig); accelConfig.HubAPIClient == nil {
			glog.V(90).Infof("failed to load provided hub kubeconfig: %v", err)
		}
	}

	return accelConfig
}

```

File: tests/accel/internal/accelinittools/accelinittools.go [hunk 10]
```diff
@@ -0,0 +1,19 @@
+package upgradeinittools
```
Comment (kononovn) [inline tests/accel/internal/accelinittools/accelinittools.go:1]: Please review suggested change below:


```
package accelinittools

import (
	"github.com/openshift-kni/eco-goinfra/pkg/clients"
	"github.com/openshift-kni/eco-gotests/tests/accel/internal/accelconfig"
	"github.com/openshift-kni/eco-gotests/tests/internal/inittools"
)

var (
	// HubAPIClient provides API access to hub cluster.
	HubAPIClient *clients.Settings
	// AccelConfig provides access to configuration parameters.
	AccelConfig *accelconfig.AccelConfig
)

func init() {
	HubAPIClient = inittools.APIClient
	AccelConfig = accelconfig.NewAccelConfig()
}
```

**Comment by kononovn** [comment]
@siserafin Please add ./test/accel/internal/accelparam 
Use this package as example: https://github.com/openshift-kni/eco-gotests/blob/main/tests/cnf/internal/cnfparams/const.go

**Comment by kononovn** [comment]
Let's build in this PR only
./tests/accel/internal/ packages.

**Review by kononovn — CHANGES_REQUESTED** [review]
Overall your PR looks good. We just need to improve test suite formatting.

File: tests/accel/upgrade/workload.go (hunk 3 above)
Comment (siserafin) [reply tests/accel/upgrade/workload.go]: Done, thanks

File: tests/accel/internal/accelconfig/config.go [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+)
+
+// ModulesConfig contains environment information related to ocp upgrade tests.
+type ModulesConfig struct {
+	PullSecret           string `envconfig:"ECO_ACCEL_UPGRADE_PULL_SECRET"`
+	Registry             string `envconfig:"ECO_ACCEL_UPGRADE_REGISTRY"`
+	UpgradeTargetVersion string `envconfig:"ECO_ACCEL_UPGRADE_UPGRADE_TARGET_IMAGE"`
+	HubKubeConfig        string `envconfig:"ECO_ACCEL_UPGRADE_HUB_KUBECONFIG"`
+	HubClusterName       string `envconfig:"ECO_ACCEL_UPGRADE_HUB_CLUSTER_NAME"`
+	HubAPIClient         *clients.Settings
```
[the remark this answers is not in the document]
Comment (siserafin) [reply tests/accel/internal/accelconfig/config.go:19]: I think I did that in internal/accelinittools/accellinittools.go. Should I move it here?

**Pull request closed without merging**